torch/csrc/Exceptions.cpp

```diff
@@ -293,6 +293,10 @@ PyObject* map_warning_to_python_type(const c10::Warning& warning) {
 /// NOLINTNEXTLINE(bugprone-exception-escape)
 PyWarningHandler::~PyWarningHandler() noexcept(false) {
   c10::WarningUtils::set_warning_handler(prev_handler_);
+  process_warnings();
+}
+
+void PyWarningHandler::process_warnings() {
```

Comment:
You need to reset the warning handler before processing the warnings to ensure no new warning is adding to the queue while you process them.

Reply:
How can there be warnings added in the loop? I don't see any calls that could and I wanted to be sure to fallback to the current reporting in case any warning pops up after.

But I guess that is still safer as `process_warnings` is to be called when no further warnings are expected/likely anyway.

Reply:
The PyErr_Warn* functions below can execute arbitrary code IIRC
You can just move the c10::WarningUtils::set_warning_handler(prev_handler_); line above here to reset the warning handler btw.

Reply:
Done that in 55c334015727d99a1ff024a10dc9c78d177254d5